sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/models/ProgressReporter.java

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callingserver.models;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * {@code ProgressReporter} offers a convenient way to add progress tracking to a given Flux.
+ */
```

Comment:
This class may need more discussion and may be better moved to core.

Reply:
Yes, the idea of moving this into core was brought by @alzimmermsft before. We discussed that this should happen after this preview, to track this separately in its own PR, given the changes to shared packages.